src/exoscale/vinyl/cursor.clj

```diff
@@ -14,23 +14,30 @@
 (defn apply-reduce
   "A variant of `RecordCursor::reduce` that honors `reduced?`.
    Hopefully https://github.com/FoundationDB/fdb-record-layer/pull/1272
-   gets in which will provide a way to do this directly from record layer."
-  [^RecordCursor cursor f init]
-  (let [acc (atom (or init (f)))]
-    (.thenApply
-     (AsyncUtil/whileTrue
-      (reify Supplier
-        (get [_]
-          (-> cursor
-              .onNext
-              (.thenApply
-               (fn/make-fun
-                (fn [^RecordCursorResult result]
-                  (let [next?   (.hasNext result)
-                        new-acc (when next? (swap! acc f (.get result)))]
-                    (and (not (reduced? new-acc)) next?))))))))
-      (.getExecutor cursor))
-     (fn/make-fun (fn [_] (unreduced @acc))))))
+   gets in which will provide a way to do this directly from record layer.
+
+   When `cont-fn` is given, it will be called on the last seen continuation
+   byte array for every new element."
+  ([^RecordCursor cursor f init cont-fn]
```

Comment:
A new arity is introduced. If non nil, `cont-f` will get called with each continuation